apps/sms/services/js/threads.js

```diff
@@ -6,16 +6,24 @@
   var threads = new Map();
   var messageMap = new Map();
 
-  function Thread(thread) {
-    var length = Thread.FIELDS.length;
-    var key;
-
-    for (var i = 0; i < length; i++) {
-      key = Thread.FIELDS[i];
-      this[key] = thread[key];
-    }
+  function extractSupportedFields(
+    { id, body, timestamp, participants, lastMessageType, status, isDraft }
+  ) {
+    return {
+      id,
+      body,
+      timestamp,
+      participants,
+      lastMessageType,
+      status,
+      isDraft: !!isDraft
```

Comment:
mmm, yeah maybe, I don't have preference. I didn't touch it in this patch because I'm not sure how we're going to handle it later - my initial idea was to replace this property with real "draft" property that will be returned from ConversationService, but it's too big change for this patch.

So just let me know if you want me to move this property to ```status``` :)